src/settings/SettingsPage.js

```diff
@@ -1,11 +1,197 @@
 import React from 'react'
 import { Helmet } from 'react-helmet'
+import { connect } from 'redux-bundler-react'
+import Box from '../components/box/Box'
+import Button from '../components/button/Button'
+import JsonEditor from './editor/JsonEditor'
+import Tick from '../icons/GlyphSmallTick'
 
-export default () => (
-  <div data-id='SettingsPage'>
-    <Helmet>
-      <title>Settings - IPFS</title>
-    </Helmet>
-    <h1>Settings</h1>
-  </div>
+const PAUSE_AFTER_SAVE_MS = 1500
+
+export class SettingsPage extends React.Component {
+  render () {
+    const {isConfigBlocked, isLoading, isSaving, hasSaveFailed, hasSaveSucceded, hasErrors, hasLocalChanges, hasExternalChanges, config, onChange, onReset, onSave, editorKey} = this.props
+    return (
+      <div data-id='SettingsPage'>
+        <Helmet>
+          <title>Settings - IPFS</title>
+        </Helmet>
+        <Box>
+          <div className='dt dt--fixed pb3'>
+            <div className='dtc v-mid'>
+              <SettingsInfo config={config} isConfigBlocked={isConfigBlocked} isLoading={isLoading} hasExternalChanges={hasExternalChanges} hasSaveFailed={hasSaveFailed} hasSaveSucceded={hasSaveSucceded} />
+            </div>
+            <div className='dtc tr v-btm pt2' style={{width: 240}}>
+              { config ? (
```

Comment:
It's better to just:

```js
{ config && <div>...</div> }
```

Reply:
@fsdiogo I prefer the look of `&&` style too, but I now always use ternary to be explicit about what I want to be rendered if the value is falsey. This is due to doing a bunch of react-native work where most falsey values other than null cause an error.